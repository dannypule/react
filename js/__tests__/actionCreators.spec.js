// @flow

import { setSearchTerm, addAPIData } from '../actionCreators';

const strangerThings = {
  title: 'The Americans',
  year: '2013â€“',
  description: 'Two Soviet intelligence agents pose as a married couple to spy on the American government.',
  poster: 'ta.jpg',
  imdbID: 'tt2149175',
  trailer: 'HjuUkbhsI24',
  rating: '8.6'
};

test('setSearchTerm', () => {
  expect(setSearchTerm('New York')).toMatchSnapshot(); // leverage snapshot ability
});

test('addAPIData', () => {
  // tests addAPIData before we test the thunk below
  // we need to make sure this is iron clad before testing the thunk
  expect(addAPIData(strangerThings)).toMatchSnapshot(); // leverage snapshot ability
});
