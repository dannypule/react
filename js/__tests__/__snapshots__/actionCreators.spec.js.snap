// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addAPIData 1`] = `
Object {
  "payload": Object {
    "description": "Two Soviet intelligence agents pose as a married couple to spy on the American government.",
    "imdbID": "tt2149175",
    "poster": "ta.jpg",
    "rating": "8.6",
    "title": "The Americans",
    "trailer": "HjuUkbhsI24",
    "year": "2013â€“",
  },
  "type": "ADD_API_DATA",
}
`;

exports[`setSearchTerm 1`] = `
Object {
  "payload": "New York",
  "type": "SET_SEARCH_TERM",
}
`;
